extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +144,272 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,Object)=} opt_processResponse An optional
+   * function to receive any response messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(ampDoc, transportOptions, opt_processResponse) {
+    const frameUrl = dev().assertString(transportOptions['iframe'],
+      'Cross-domain frame parameters missing ${this.type_}');
+    this.beginUsingCrossDomainIframe_(ampDoc, frameUrl,
+      transportOptions['extraData']);
+    const frameData = Transport.crossDomainIframes_[frameUrl];
+    const iframeMessagingClient = frameData.iframeMessagingClient;
+    iframeMessagingClient.registerCallback(
+      MessageTypes.ampAnalytics3pReady, () => {
+        iframeMessagingClient.setHostWindow(frameData.frame.contentWindow);
+        Transport.setIsReady_(frameUrl);
+      });
+    iframeMessagingClient.registerCallback(
+      MessageTypes.ampAnalytics3pResponse, response => {
+        dev().assert(response &&
+          response[MessageTypes.ampAnalytics3pResponse],
+          'Received empty response from 3p analytics frame');
+        if (!opt_processResponse) {
+          dev().warn(TAG_, 'Received response from 3p analytics frame when' +
+            ' none was expected');
+        }
+        opt_processResponse(this.type_,
+          response[MessageTypes.ampAnalytics3pResponse]);
+      });
+  }
+
+  /**
+   * If iframe doesn't exist for this iframe url, create it.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!string} frameUrl The URL of the frame to send the data to
+   * @param {string=} opt_extraData The data to send to the frame
+   * @private
+   */
+  beginUsingCrossDomainIframe_(ampDoc, frameUrl, opt_extraData) {
+    if (Transport.hasCrossDomainIframe_(frameUrl)) {
+      Transport.incrementIframeUsageCount_(frameUrl);
+      this.informCrossDomainIframeOfNewCreative_(frameUrl, opt_extraData);
```

Comment:
I considered doing exactly that, but note that this would reverse the order of this.informCrossDomainIframeOfNewCreative_() and frame.src = frameUrl, in createCrossDomainIframe_. I was concerned about creating a race condition in case the frame rendered immediately between the two commands. What do you think?